packages/calling-stateful-client/src/StreamUtils.ts

```diff
@@ -468,21 +470,46 @@ function disposeViewLocalVideo(context: CallContext, internalContext: InternalCa
     return;
   }
 
-  // Sets the status and also renderer. I think we need to always set renderer to undefined since in all status when
-  // cleaned up should have renderer as undefined. If the status is 'Rendered' and renderer is not defined it should
-  // be cleaned up below so we can set it to undefined here.
-  if (renderInfo.status === 'Rendering') {
+  // Nothing to dispose of or clean up -- we can safely exit early here.
+  if (renderInfo.status === 'NotRendered') {
+    _logEvent(callingStatefulLogger, {
+      name: EventNames.LOCAL_STREAM_ALREADY_DISPOSED,
+      level: 'info',
+      message: 'LocalVideoStream is already disposed.'
+    });
+    return;
+  }
+
+  // Status is already marked as "stopping" so we can exit early here. This is because stopping only occurs
+  // when the stream is being created in createView but hasn't been completed being created yet. The createView
+  // method will see the "stopping" status and perform the cleanup
+  if (renderInfo.status === 'Stopping') {
     _logEvent(callingStatefulLogger, {
       name: EventNames.LOCAL_STREAM_STOPPING,
       level: 'info',
-      message: 'Local stream is still rendering. Changing status to stopping.',
+      message: 'Local stream is already stopping.',
       data: streamLogInfo
     });
-    internalContext.setLocalRenderInfo(callId, renderInfo.stream, 'Stopping', undefined);
-  } else {
-    internalContext.setLocalRenderInfo(callId, renderInfo.stream, 'NotRendered', undefined);
+    return;
+  }
+
+  // If the stream is in the middle of being rendered (i.e. has state "Rendering"), we need the status as
+  // "stopping" without performing any cleanup. This will tell the `createView` method that it should stop
+  // rendering and clean up the state once the view has finished being created.
+  if (renderInfo.status === 'Rendering') {
+    _logEvent(callingStatefulLogger, {
+      name: EventNames.REMOTE_STREAM_STOPPING,
+      level: 'info',
+      message: 'Remote stream is still rendering. Changing status to stopping.',
+      data: streamLogInfo
+    });
+    internalContext.setLocalRenderInfo(callId, renderInfo.stream, 'Stopping', renderInfo.renderer);
```

Comment:
Looking at the createView method, wouldn't this set the state to "Rendering" again?

Reply:
If it's called again, yep!

For context for this message, their are four renderInfo states: `NotRendered`, `Rendering`, `Stopping`, `Rendered`.
NotRendered is the starting state or end of the cleanup state
Rendering is set when createView is asynchronously creating the render stream
Stopping is set when disposeView is called, but createView is currently asynchronously creating the view so it can't be cleaned up right now.
Rendered is the happy path when the render has finished.

How DisposeView works:
- will try to dispose the view.
- if the view is NotRendered or Stopping, it can no-op and exit early  -- nothing can be done
- if the view is Rendering, it can't dispose the view right now, so it sets the state to Stopping (the createView then checks for this when it finishes creating the view and performs the cleanup instead of disposeView)
- if the view is rendered, this is the happy path and it will do the cleanup

How createView works:
- if the view is Rendered or Rendering it can simply no-op -- nothing can be done
- if the view is in Stopping, it is only in this state because another createView is asynchronously waiting for the view to be completed and a disposeView was subsequently called that set it to stopping. Now that createView has been called again it can change it back to Rendering safely.
- If the view is in NotRendered this is the happy path and it can create the view
	- Now it creates the view but this is asynchronous
	- Once it completes the view if the state is stopping, it should clean up and dispose of the newly created view
	- Otherwise it's happy path time again and can set the state as Rendered 


now for your question
> Looking at the createView method, wouldn't this set the state to "Rendering" again?

Yes! If, and only if, createView is called again before the previous createView's asynchronous view creation has completed